codex-rs/core/src/rollout.rs

```diff
@@ -204,6 +205,21 @@ impl RolloutRecorder {
         info!("Resumed rollout successfully from {path:?}");
         Ok((Self { tx }, saved))
     }
+
+    pub async fn shutdown(&self) -> std::io::Result<()> {
```

Comment:
I know this has been a long review, but I feel much better that the changes to `RolloutRecorder.rs` are much smaller now.